internal/operators/ocs/ocs_operator.go

```diff
@@ -92,6 +95,49 @@ func (o *operator) GetMonitoredOperator() *models.MonitoredOperator {
 }
 
 // GetHostRequirements provides operator's requirements towards the host
-func (o *operator) GetHostRequirements(context.Context, *common.Cluster, *models.Host) (*models.ClusterHostRequirementsDetails, error) {
-	return &models.ClusterHostRequirementsDetails{}, nil
+func (o *operator) GetHostRequirements(_ context.Context, cluster *common.Cluster, host *models.Host) (*models.ClusterHostRequirementsDetails, error) {
+	numOfHosts := len(cluster.Hosts)
+
+	if numOfHosts < 3 {
+		return nil, fmt.Errorf("number of hosts is less than 3")
+	} else if numOfHosts == 3 { // Compact Mode
+		if host.Role == models.HostRoleMaster || host.Role == models.HostRoleAutoAssign {
+			return &models.ClusterHostRequirementsDetails{
+				CPUCores:   CPUCompactMode,
+				RAMMib:     conversions.GbToMib(MemoryCompactMode),
+				DiskSizeGb: MinDiskSize,
+			}, nil
+		}
+		return nil, fmt.Errorf("unsupported role: %s", host.Role)
+	}
+
+	if host.Role == models.HostRoleAutoAssign { //If auto-assign we cannot proceed with minimal or standard mode
+		return nil, fmt.Errorf("unsupported role: %s", host.Role)
+	}
+
+	// Minimum mode is the minimum version of the standard mode. If the resources will be more, standard mode will be deployed automatically.
+	// In minimum and standard mode we don't care about masters so returning zero
+	if host.Role == models.HostRoleMaster {
+		return &models.ClusterHostRequirementsDetails{CPUCores: 0, RAMMib: 0}, nil
+	}
+
+	inventory, err := hostutil.UnmarshalInventory(host.Inventory)
+
+	if err != nil {
+		return nil, err
+	}
+
+	disks := int64(getValidDiskCount(inventory.Disks) - 1)
+	if disks > 0 {
```

Comment:
The disk check should also be added for compact mode. If no disk is present in compact mode, then return error.

Reply:
Also add test accordingly.